api-server/api-server-common/src/storage/impls/postgres/transactional/mod.rs

```diff
@@ -107,43 +126,60 @@ impl<'a> ApiServerPostgresTransactionalRo<'a> {
         &mut self,
         block_id: Id<Block>,
     ) -> Result<Option<BlockAuxData>, ApiServerStorageError> {
-        let mut conn = QueryFromConnection::new(&self.connection);
+        let mut conn = QueryFromConnection::new(self.connection.as_ref().expect(CONN_ERR));
         let res = conn.get_block_aux_data(block_id).await?;
 
         Ok(res)
     }
 }
 
 pub struct ApiServerPostgresTransactionalRw<'a> {
-    connection: PooledConnection<'a, PostgresConnectionManager<NoTls>>,
+    // Note: This is an Option due to needing to pry the connection out of Self in Drop
+    connection: Option<PooledConnection<'static, PostgresConnectionManager<NoTls>>>,
     finished: bool,
+    db_tx_sender: tokio::sync::mpsc::UnboundedSender<
+        PooledConnection<'static, PostgresConnectionManager<NoTls>>,
+    >,
+    // Note: This exists to enforce that a transaction never outlives the database object,
+    //       given that all connections have 'static lifetimes
+    _marker: std::marker::PhantomData<&'a ()>,
 }
 
 impl<'a> Drop for ApiServerPostgresTransactionalRw<'a> {
     fn drop(&mut self) {
         if !self.finished {
```

Comment:
Since getting a connection has been change from `get()` to `get_owned()`, should we always `send()` the connection back to the poll regardless of `finished`?